src/middleware/error_handling.ts

```diff
@@ -46,13 +47,21 @@ export function generateError(err: Error): ErrorBodyWithHTTPStatusCode {
                 },
             };
         }
+    } else if ((err as any).statusCode) {
+        return {
+            statusCode: HttpStatus.BAD_REQUEST,
+            errorBody: {
+                reason: err.message,
+            },
```

Comment:
Don't fully understand this reasoning.

Reply:
Previously:

```
if (isAPIError OR err.statusCode) {
    // generateError
    if (isAPIError) {
        do x;
    } else { // effectively means (if err.statusCode)
        do y; // generic HttpStatus.BAD_REQUEST
    }
}
```

Now:
```
if (isAPIerror) {
    do x;
} else if (err.statusCode) {
    do y;
} else {
    return generic 500;
}
```

So it's the same as before, but with the added fallback case if generic 500.

Reply:
discussed on Slack. We realised the existing code was borked. If an error has a statuscode then it's a named error and the statuscode should be preserved. Otherwise it's a generic error and should be treated as a generic 500.